### DISCUSSION TITLE
Extruding polygon from heightmap

### DISCUSSION THREAD

**Discussion by 122-TheodoreLEHOUX** [tylermorganwall/rayshader#267] [Q&A]
I was wondering if it was possible to render polygon with the bottom on the heightmap ?

**Comment by Witellsteim** [comment]
I think you mean if it is possible to render polygons on the surface? I was wondering the exact same.

**Reply by 122-TheodoreLEHOUX** [reply]
Yes sorry that what I was meaning. I tried to retrieve the elevation of the map at the polygon position to use the data_column_bottom argument but elevation is not corresponding saddly.

**Reply by Witellsteim** [reply]
I tried the same. The argument only seems to influence the height the polygons are extruded downwards. So they end up upside down so to say, while still bein rendered from the same baseline

**Reply by 122-TheodoreLEHOUX** [reply]
Find a solution : 

`bottom_elevation<- function(dataframe, output) {
    coo <- st_coordinates(st_geometry(dataframe$geometry))[2, ]
    long <- coo[1]
    lat <- coo[2]
    coords <- c(long, lat)
    # Use the extract function to retrieve the elevation data, localtif is tif file rasterized
    bottom <- extract(localtif, cbind(coords[1], coords[2]))
    bottom <- bottom / estimated_zscale
    return(bottom)
}`

And then apply it to your dataframe : 

`bottom_column<- apply(your_dataframe, 1, bottom_elevation)
your_dataframe<- cbind(your_dataframe, bottom = bottom_column)
render_polygons(
    your_dataframe,
    extent = extent_zoomed, heightmap = elmat, data_column_bottom = "bottom "
)
`

**Reply by Witellsteim** [reply]
I am going over your code. So you want to extract the height at the location of the polygon and then add this as a new value for the "data_column_bottom" argument to the dataframe containing the polygon. Unfortunately this does not resolve my problem. The Polygons are still bein extruded downwards and dont stand on the surface.
Furthermore (this has nothing to do with your code) i observed, that some Polygons with a greater value for their height are being render from the soliddepth upwards. This seems highly random...
